### DISCUSSION TITLE
Passport Strategy: Additional parameters for the authorization request

### DISCUSSION THREAD

**Discussion by warrenwingaru** [panva/openid-client#735] [Q&A] [answered]
As I understand it, in order to always show a login prompt is to include a prompt: 'login' in the authorization request. After checking the source code for the passport implementation

```js
  // prettier-ignore
  /**
   * Return extra parameters to be included an authorization request.
   */
  authorizationRequestParams<
    TOptions extends
      passport.AuthenticateOptions = passport.AuthenticateOptions,
  >(
    // @ts-ignore
    req: express.Request, options: TOptions,
  ): URLSearchParams | Record<string, string> | undefined {
    return {}
  }
```
Is this currently being worked on? or am I missing something?

## Example
```js
app.get('/login', passport.authenticate(server.hostname, { prompt: 'login' }))
```

Edit: Add example usage

**Comment by panva** [comment, marked as answer]
It was not currently worked on, no, but I missed that @types/passport and apparently even the passport docs mention `scope` and `prompt` which is what lead you to believe it would be handled out of the box, despite that not being the case in e.g. https://github.com/jaredhanson/passport-oauth2

Nevertheless, I've updated the strategy in https://github.com/panva/openid-client/releases/tag/v6.1.6 to use the options (scope, prompt, and passReqToCallback). But the intention remains that for anything non-passport you need to overload this method to take effect.

**Reply by warrenwingaru** [reply]
Thanks for the reply, it flew over my head to just overload the method.

**Reply by panva** [reply]
Can you confirm that with v6.1.6 you don't need to overload to use prompt?

**Reply by warrenwingaru** [reply]
Will do when I get home, I'll let you know the soonest

**Reply by warrenwingaru** [reply]
Yup, all good now. Thanks
